cloudinit/net/dhcp.py

```diff
@@ -41,7 +41,9 @@ class NoDHCPLeaseMissingDhclientError(NoDHCPLeaseError):
     """Raised when unable to find dhclient."""
 
 
-def maybe_perform_dhcp_discovery(nic=None, dhcp_log_func=None):
+def maybe_perform_dhcp_discovery(
+    nic=None, dhcp_log_func=None, alt_exe_dir=None
```

Comment:
Perhaps  needs_exe=True to temp_utils.tmpdir() should just do the right thing and avoid requiring alt_exe_dir passed around?

Reply:
I was wondering about that too (or some other way to reduce the number of repeated changes), however doing that would couple `temp_utils` to `distro`. Is maybe that why @aciba90 didn't do this originally? I don't know off the top of my head if such a change would cause circular import issues, but it might.

Reply:
Refactored to expose the target tmp_dir and let callers (datasources, modules) determine if the dir is mounted in a noexec point and give an exec dir via the distro in use.